src/components/HomeView.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { Button } from '@mui/material';
+import { AddCircle } from '@mui/icons-material';
+import { createStyles, WithStyles, withStyles } from '@mui/styles';
+import { CustomTheme } from '@mui/material/styles';
+import { RouterLocation, RouterRootState } from 'connected-react-router';
+import { LocationState } from 'history';
+import DatasetView from './DatasetView';
+import SnapshotView from './SnapshotView';
+import JobView from './JobView';
+import SearchTable from './table/SearchTable';
+import { TdrState } from '../reducers';
+
+const styles = (theme: CustomTheme) =>
+  createStyles({
+    pageRoot: {
+      padding: '16px 24px',
+      display: 'flex',
+      flexDirection: 'column',
+      height: '100%',
+    },
+    header: {
+      alignItems: 'center',
+      color: theme.typography.color,
+      display: 'flex',
+      fontWeight: 500,
+      fontSize: 16,
+      height: 21,
+      letterSpacing: 1,
+    },
+    jadeTableSpacer: {
+      paddingBottom: theme.spacing(12),
+    },
+    jadeLink: {
+      ...theme.mixins.jadeLink,
+      float: 'right',
+      fontSize: 16,
+      fontWeight: 500,
+      height: 20,
+      letterSpacing: 0.3,
+      paddingLeft: theme.spacing(4),
+      paddingTop: theme.spacing(4),
+    },
+    title: {
+      color: theme.palette.secondary.dark,
+      fontSize: '1.5rem',
+      fontWeight: 700,
+      flex: '1 1 0',
+      'padding-right': '2em',
+      display: 'flex',
+    },
+    titleAndSearch: {
+      display: 'flex',
+      'margin-top': '1.25em',
+      'margin-bottom': '1.25em',
+    },
+    headerButton: {
+      padding: 10,
+      marginLeft: 30,
+      textTransform: 'none',
+    },
+    buttonIcon: {
+      marginRight: 6,
+      fontSize: '1.5rem',
+    },
+  });
+
+interface IProps extends WithStyles<typeof styles> {
+  location: RouterLocation<LocationState>;
+}
+
+function HomeView({ classes, location }: IProps) {
+  const [searchString, setSearchString] = useState('');
+  const prefixMatcher = /\/[^/]*/;
+  const tabValue = prefixMatcher.exec(location.pathname)?.[0];
+
+  let pageTitle = 'Terra Data Repository';
+  let searchable = true;
+  let tableValue = <div />;
+  if (tabValue === '/datasets') {
+    pageTitle = 'Datasets';
+    tableValue = <DatasetView searchString={searchString} />;
+  } else if (tabValue === '/snapshots') {
+    pageTitle = 'Snapshots';
+    tableValue = <SnapshotView searchString={searchString} />;
+  } else if (tabValue === '/activity') {
+    pageTitle = 'Activity';
+    searchable = false;
+    tableValue = <JobView searchString={searchString} />;
+  }
```

Comment:
I think the best practice here would be to use react state and hooks to manage these values. I would think we would need this in the case that the location changed, but clearly, it works without it, so I think I'm just rusty on what happens without react hooks 😅 . At any rate, I still think using react state/hooks is a pattern that I'd like to see us use. 
```suggestion
  const [searchString, setSearchString] = useState('');
  const prefixMatcher = /\/[^/]*/;
  const [tabValue, setTabValue] = useState(prefixMatcher.exec(location.pathname)?.[0]);
  const [pageTitle, setPageTitle] = useState('Terra Data Repository');
  const [searchable, setSearchable] = useState(true);
  const [tableValue, setTableValue] = useState(<div />);
  
 useEffect(() => {
     setTabValue(prefixMatcher.exec(location.pathname)?.[0]);
   }, [location.pathname]);
 
 useEffect(() => {
     if (tabValue === '/datasets') {
         setPageTitle('Datasets');
         setTableValue(<DatasetView searchString={searchString} />);
         setSearchable(true);
      } else if (tabValue === '/snapshots') {
          setPageTitle('Snapshots');
          setTableValue(<SnapshotView searchString={searchString} />);
          setSearchable(true);
       } else if (tabValue === '/activity') {
          setPageTitle(' 'Activity');
          setSearchable(false);
          setTableValue(<JobView searchString={searchString} />);
        }
  }, [tabValue, pageTitle, tableValue, searchable]);
```

Reply:
Thanks for this suggestion! I was porting over what previously existed but it seemed a little funny to me, I'll give this a shot.

Reply:
Discussed this offline - this is getting out of scope, so disregard this comment.